### PR TITLE
Add PDF Generation and update OpenAPI spec

### LINKED ISSUE #90: Re-add PDF generation
https://stackoverflow.com/questions/54259816/how-to-generate-a-pdf-or-markup-from-openapi-3-0

This is the old make target for PDF generation of the mkdocs-based site:

```Makefile
pdf: build
	wkhtmltopdf --version;
	# fix protocol relative URLs, see https://github.com/wkhtmltopdf/wkhtmltopdf/issues/2713
	find site/ -type f -name '*.html' | xargs sed -i 's|href="//|href="https://|g'
	find site/ -type f -name '*.html' | xargs sed -i 's|src="//|src="https://|g'
	wkhtmltopdf --margin-top 20mm --no-background --javascript-delay 5000 \
	file://$(shell pwd)/site/index.html \
	file://$(shell pwd)/site/compendium/view/index.html \
	file://$(shell pwd)/site/compendium/candidate/index.html \
	file://$(shell pwd)/site/compendium/files/index.html \
	file://$(shell pwd)/site/compendium/delete/index.html \
	file://$(shell pwd)/site/compendium/upload/index.html \
	file://$(shell pwd)/site/compendium/public_share/index.html \
	file://$(shell pwd)/site/compendium/download/index.html \
	file://$(shell pwd)/site/compendium/metadata/index.html \
	file://$(shell pwd)/site/compendium/substitute/index.html \
	file://$(shell pwd)/site/compendium/link/index.html \
	file://$(shell pwd)/site/job/index.html \
	file://$(shell pwd)/site/search/index.html \
	file://$(shell pwd)/site/shipment/index.html \
	file://$(shell pwd)/site/user/index.html \
	o2r-web-api.pdf
```

### LINKED ISSUE #92: Transform bindings spec to OpenAPI
See #82 (now closed)

@Fmazin please check with @NJaku01 if there is anything missing.

### LINKED ISSUE #94: Check if timestamps are correctly documented
See branch https://github.com/o2r-project/api/tree/timestamps (last two commits).

If everything is in the OpenAPI spec, then tell @nuest the branch `timestamps` can be deleted.

### LINKED ISSUE #97: Switch from ISO-8601 to RFC-3339
Instead of ISO 8601, we should refer (and use) only to RFC-3339 (tools.ietf.org/html/rfc3339) which is open/free.

This only requires to change the name of the standard (search project for "8601").

### PR DESCRIPTION
**Pull request by Fmazin**
This PR will resolve #90, fix #92, resolve #94 and close #97.

A PDF version of the OpenAPI specification will now be created every time the repository is changed. A preview to the pdf can be viewed [here](https://fmazin.github.io/api/docs/pdf/o2r-openapi.pdf).

In addition to this, bindings are now in the spec. Timestamps have also been reworked.

### CODE REVIEW

File: docs/o2r-openapi.yml [hunk 1]
```diff
@@ -1301,7 +1325,115 @@ paths:
...
+  #Bindings Paths
+  /bindings/binding:
+    post:
+      tags:
+        - Bindings
+      summary: Create new binding
+      operationId: upload_binding
+      requestBody:
+        description: |
+          The Bindning to be uploaded.
```
Comment (nuest) [inline docs/o2r-openapi.yml:1338]: Typo, "Binding".

File: docs/o2r-openapi.yml [hunk 2]
```diff
@@ -1301,7 +1325,115 @@ paths:
...
+        200:
+          description: Binding Upload response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/upload_binding_response'
+              examples:
+                binding_success:
+                  $ref: '#/components/examples/binding_success'
+#
```
Comment (nuest) [inline docs/o2r-openapi.yml:1358]: If it's deprecated, can we not remove this function?

**Review by nuest — CHANGES_REQUESTED** [review]
- Fix title for publishment, delete shipment... there are a bunch of headlines with `_` underscores - is this correct? `create_job`, `list_job`, `update_metadata`, `shipment_status`, `publish_deposition`, `xyz_binding`, `list_user`, ...
  image
  
  image
  
  image

- Fix problem with bold (?) font (in a headline?)
  image
  
  image

- Fix typo "list_reciepients"

---------

I have put some things into this issue - we don't need to fix everything now but can iterate later: https://github.com/o2r-project/api/issues/99

**Review by Fmazin — COMMENTED** [review]
I have removed the extract binding section and fixed the bold sections. I have done the latter with the command line editor from perl which allows for rather simple and elegant regex search and replace. The headings with `_` are still a Problem but i will probably use also some regex.

**Comment by nuest** [comment]
@Fmazin let me know when I can take another look.

Also, I guess it would be wise to work in branches that are not `master` from now on to avoid PDF generation while developing the PR, see [`9ea1c10` (#98)](https://github.com/o2r-project/api/pull/98/commits/9ea1c101dd6c3e6b36abae2e970604c683331393)

image

**Comment by nuest** [comment]
@Fmazin Ready to be merged?

**Comment by Fmazin** [comment]
Yes this is ready to be merged.

**Review by nuest — APPROVED** [review]

**Pull request merged**